tests/base/src/test/java/org/keycloak/tests/admin/SMTPConnectionTest.java

```diff
@@ -220,9 +225,54 @@ public void testWithTokenAuthEnabledAndRetryWithValidTokenInSecondTry() throws E
 
     }
 
+    @Test
+    @Order(9)
+    public void testAllowUTF8() throws Exception {
+        // utf8 on from not allowed if allowutf8 not enabled
+        Response response = adminClient.realms().realm(managedRealm.getName()).testSMTPConnection(settings("127.0.0.1", "3025", "autoñ@keycloak.org",
+                null, null, null, null, null));
+        assertStatus(response, 500);
+
+        // utf-8 on from but in domain part is allowed and transformed
```

Comment:
Nitpick: These seem to be separate test cases. Perhaps it would be useful to put them in separate test methods? Maybe we could even execute them in parallel?

Reply:
This approach works for me as well. Resolving.